src/build_vehicle_gui.cpp

```diff
@@ -1590,8 +2134,40 @@ struct BuildVehicleWindow : Window {
 	void OnQueryTextFinished(char *str) override
 	{
 		if (str == nullptr) return;
-
-		DoCommandP(0, this->rename_engine, 0, CMD_RENAME_ENGINE | CMD_MSG(STR_ERROR_CAN_T_RENAME_TRAIN_TYPE + this->vehicle_type), nullptr, str);
+		VehicleFilter& filter = _engine_sort_filter[this->vehicle_type];
+
+		switch (widget_for_query)
+		{
+		case WID_BV_RAIL_SPEED_EDIT:
```

Comment:
Case labels should be indented a level deeper than the switch statement, so the code inside each case is indented two levels deeper than the switch statement.